src/containers/monitor.jsx

```diff
@@ -250,6 +250,7 @@ Monitor.propTypes = {
     resizeMonitorRect: PropTypes.func.isRequired,
     spriteName: PropTypes.string, // eslint-disable-line react/no-unused-prop-types
     targetId: PropTypes.string,
+    toolboxXML: PropTypes.string, // eslint-disable-line react/no-unused-prop-types
```

Comment:
I am not aware of anything exactly like that. I suppose if we were using hooks we could do something like:

```js
const toolboxXML = useSelector(...);
const label = useMemo(functionToCalculateLabel, [toolboxXML, otherDependencies]);
```

That way `toolboxXML` is used in the dependencies array, but I think there is a eslint test that could complain about the fact that `toolboxXML` was not actually used in the memoized function.

I agree having an unused prop feels a bit wrong. Seems like the heart of the problem is that we are computing labels based on some internal state that the vm is holding. React is good at rendering updates when props or state changes. I'm thinking about the idea of the monitor labels being stored in a React or Redux state similar to (or combined with) `state.scratchGui.monitorLayout` or `state.scratchGui.monitors`. I can't come up with a solution I like more than using `toolboxXML` updates.

 ---

Regarding adding a test that will fail without this change - With some minor tweaks, I could test something like:

```js
expect(Monitor.propTypes).toHaveProperty('toolboxXML');
```

That would make sure someone doesn't remove that propType.

Or we could test that when you render the component without the `toolboxXML` prop it has a warning. Unfortunately it looks like this is only displayed in the console, so I've seen examples of mocking `console.error` and trying to match the error message. Listening to `console.error` seems like a weird solution, but I can't find a better suggestion.

Neither stops someone from breaking things by removing line 270. Not a test, but another option is to just move the comment from line 269 to above this line. You'd probably notice the comment before attempting to clean it up. What do you think about those ideas, or is there another one you are thinking of?

Reply:
Hmm, can't we depend on the locale (`state.locales.locale`) instead of `toolboxXML`? That does change between ja and ja-hira. And having a big string as a props, doesn't sound great IMO.

Reply:
Totally agree that depending on `locale` seems like a better option, and I tried that first. However, the virtual machine does not seem to refresh the locale synchronously when it changes (`extension-manager.js : refreshBlocks()`). This means that Monitor can re-render before the virtual machine has updated all of the blocks with the new locale. When this happens, the monitor still has the label for the previous locale, not the new one.

I do want to be careful about the size of `toolboxXML`. I may be missing something, but I can't spot a problem. I don't think the component will be creating a copy of the `toolboxXML` contents from the redux store, just a reference to it. React-redux will check to see if the value changed, but it should be using [strict equality for that comparison](https://react-redux.js.org/using-react-redux/connect-mapstate#return-values-determine-if-your-component-re-renders). I can't think of a scenario where this prop is using significant memory or computation time.

Reply:
@cwillisf - I added an integration test for this issue.